util/logtools/anonymizer.ts

```diff
@@ -242,6 +264,70 @@ export default class Anonymizer {
     return splitLine.join('|');
   }
 
+  // This method assumes that repeating fields use the current `CombatantMemory` structure
+  // of key/value pairs, e.g. Key1|Value1|...|KeyN|ValueN|.
+  // If other structures are used, this method will need to be updated.
+  private anonymizeRepeatingFields(
+    rfToAnonymize: { [key: string | number]: string | number | null },
+    splitLine: string[],
+    startIdx: number,
+    notifier: Notifier,
+  ): string[] {
+    for (const [idFieldKey, nameFieldKey] of Object.entries(rfToAnonymize)) {
+      let idFieldKeyIdx: number;
+      let idIsRepeatingField = true; // default assumption
+
+      if (typeof idFieldKey === 'number' || !isNaN(parseInt(idFieldKey))) {
+        idIsRepeatingField = false;
+        idFieldKeyIdx = parseInt(idFieldKey);
+      } else
+        idFieldKeyIdx = splitLine.findIndex((field, idx) =>
+          idx >= startIdx && field === idFieldKey
+        );
+
+      // ID field is not present; not an error condition for repeating fields.
+      if (idFieldKeyIdx === -1)
+        continue;
+
+      const idFieldValueIdx = idIsRepeatingField ? idFieldKeyIdx + 1 : idFieldKeyIdx;
+      const idFieldValue = (splitLine[idFieldValueIdx] ?? '').toUpperCase();
+
+      // If not a playerId value, ignore.
+      // TODO: Could add warnings here for malformed values, but there are many cases
+      // where it's fine if it's not a playerId, so ignoring seems safer/less noisy.
```

Comment:
I would suggest updating this comment in general. Basically, `OverlayPlugin` only models a full `BattleChara` object, but other actors end up in the same array and OverlayPlugin just treats them all as the same. So junk data can end up in fields that aren't valid for other actors. 

You can see, based on the `BattleChara` class def from FFXIVClientStructs:
```cs
// Client::Game::Character::BattleChara
//   Client::Game::Character::Character
//     Client::Game::Object::GameObject
//     Client::Graphics::Vfx::VfxDataListenner
//     Client::Game::Character::CharacterData
```

However, any `Client::Game::Object::GameObject` can end up in this array.

So instead of `but there are many cases where it's fine if it's not a playerId, so ignoring seems safer/less noisy.`, I'd suggest wording more like `but due to the nature of ``CombatantMemory`` data from OverlayPlugin, any ``GameObject`` could be returned as a line with malformed data if it's not a full combatant`, or something along those lines.